### PR TITLE
Fix deadlock in TestHttpsAppProbe test

### PR DESCRIPTION
**Pull request by howardjohn**
The problem here is we had a `for {}` loop that was constantly locking and unlocking the mutex, starving the other thread that need to acquire the lock to make progress (ref https://medium.com/a-journey-with-go/go-mutex-and-starvation-3f4f4e75ad50). This change just adds some backoff + timeout which allows this to complete successfully

Example failure: https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17253/unit-tests_istio/706

### CODE REVIEW

File: pilot/cmd/pilot-agent/status/server_test.go [hunk 1]
```diff
@@ -183,10 +185,16 @@ func TestHttpsAppProbe(t *testing.T) {
 	go server.Run(context.Background())
 
 	var statusPort uint16
-	for statusPort == 0 {
+	if err := retry.UntilSuccess(func() error {
```
Comment (rlenglet) [inline pilot/cmd/pilot-agent/status/server_test.go]: Can't you just add a channel into `Server` that is closed when the `statusPort` is set?
Would that be overkill?
```
type Server struct {
	...
	statusPort          uint16
	statusPortSet       chan struct{}
	...
}
```

```
func (s *Server) Run(ctx context.Context) {
	...
		s.mutex.Lock()
		s.statusPort = uint16(allocatedPort)
		s.mutex.Unlock()
		close(s.statusPortSet)
	...
}
```

And here:
```
	select {
	case <-server.statusPortSet:
	case <-time.After(30*time.Second):
		t.Fatal("failed to getport after 30 seconds")
	}
	...
```

Comment (howardjohn) [reply pilot/cmd/pilot-agent/status/server_test.go]: That is probably a safer fix but more invasive to the application code just for the sake of testing. I'm open to either solution though

**Pull request merged**